src/universal/components/RetroReflectPhase/RetroReflectPhase.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Renders the UI for the reflection phase of the retrospective meeting
+ *
+ * @flow
+ */
+import type {Team} from 'universal/types/schema.flow';
+
+import React from 'react';
+import styled from 'react-emotion';
+import {createFragmentContainer} from 'react-relay';
+
+import ReflectionTypeColumn from './ReflectionTypeColumn';
+
+type Props = {
+  team: Team,
+};
+
+const ReflectPhaseWrapper = styled('div')({
+  height: '100%',
+  display: 'flex',
+  justifyContent: 'space-around',
+  width: '100%'
+});
+
+const RetroReflectPhase = ({team}: Props) => {
+  const {newMeeting, meetingSettings} = team;
+  if (!meetingSettings.hasOwnProperty('phaseItems')) {
+    return null;
+  }
+  // The nested union types of meetingSettings/phaseItems are creating a multiplicative
+  // effect on the possible values of meetingSettings, making it really difficult
+  // $FlowFixMe
+  const {phaseItems} = meetingSettings;
+  return (
+    <ReflectPhaseWrapper>
+      {phaseItems && phaseItems.map((phaseItem) =>
+        <ReflectionTypeColumn newMeeting={newMeeting} key={phaseItem.id} retroPhaseItem={phaseItem} team={team} />
+      )}
+    </ReflectPhaseWrapper>
+  );
+};
+
+export default createFragmentContainer(
+  RetroReflectPhase,
+  graphql`
+    fragment RetroReflectPhase_team on Team {
+      ...ReflectionTypeColumn_team
+      meetingSettings(meetingType: $meetingType) {
+        ... on RetrospectiveMeetingSettings {
+          phaseItems {
+            ... on RetroPhaseItem {
+              id
+              ...ReflectionTypeColumn_retroPhaseItem
```

Comment:
oh nice you did find it! using this we should be able to get rid of that switch statement above